src/main/java/frc/team2412/robot/subsystems/DrivebaseSubsystem.java

```diff
@@ -20,145 +16,190 @@
+						new Translation2d(Units.inchesToMeters(8.5), Units.inchesToMeters(-8.5)), // front right
+						new Translation2d(Units.inchesToMeters(-8.5), Units.inchesToMeters(8.5)), // back left
+						new Translation2d(Units.inchesToMeters(-8.5), Units.inchesToMeters(-8.5)) // back right
+					};
 
 	SwerveDriveKinematics kinematics =
 			new SwerveDriveKinematics(
 					moduleLocations[0], moduleLocations[1], moduleLocations[2], moduleLocations[3]);
 
-	private AHRS gyroscope;
+	private Gyroscope gyroscope;
 
 	private SwerveDriveOdometry odometry;
 	private Pose2d pose;
 
 	private Field2d field = new Field2d();
 
 	public DrivebaseSubsystem() {
-		gyroscope = new AHRS(SerialPort.Port.kMXP);
+		gyroscope = IS_COMP ? new Pigeon2Gyro(Hardware.GYRO_PORT) : new NavXGyro(SerialPort.Port.kMXP);
 
-		odometry =
-				new SwerveDriveOdometry(
-						kinematics, Rotation2d.fromDegrees(gyroscope.getYaw()), getModulePositions());
+		odometry = new SwerveDriveOdometry(kinematics, gyroscope.getAngle(), getModulePositions());
 		pose = odometry.getPoseMeters();
 
 		balanceController =
-				PFFController.ofDouble(tipF, tipP)
-						.setTargetPosition((double) gyroscope.getRoll())
-						.setTargetPositionTolerance(tipTolerance);
+				PFFController.ofDouble(TIP_F, TIP_P)
+						.setTargetPosition(gyroscope.getRawRoll().getDegrees())
+						.setTargetPositionTolerance(TIP_TOLERANCE);
 
 		// configure encoders offsets
 		for (int i = 0; i < moduleEncoders.length; i++) {
 			moduleEncoders[i].configFactoryDefault();
+			moduleEncoders[i].configSensorInitializationStrategy(
+					SensorInitializationStrategy.BootToAbsolutePosition);
 		}
 
 		// configure drive motors
 		for (int i = 0; i < moduleDriveMotors.length; i++) {
-			WPI_TalonFX driveMotor = moduleDriveMotors[i];
-			driveMotor.setNeutralMode(NeutralMode.Brake);
-			driveMotor.setSensorPhase(true);
-
-			driveMotor.configNeutralDeadband(0.01);
-			driveMotor.configSelectedFeedbackSensor(
-					TalonFXFeedbackDevice.IntegratedSensor, talonFXLoopNumber, canTimeoutMS);
-
-			driveMotor.config_kP(talonFXLoopNumber, 0.1);
-			driveMotor.config_kI(talonFXLoopNumber, 0.001);
-			driveMotor.config_kD(talonFXLoopNumber, 1023.0 / 20660.0);
+			MotorController driveMotor = moduleDriveMotors[i];
+			driveMotor.setNeutralMode(MotorController.MotorNeutralMode.BRAKE);
+
+			if (IS_COMP) {
+				driveMotor.setControlMode(MotorControlMode.VOLTAGE);
+			} else {
+				driveMotor.setControlMode(MotorControlMode.VELOCITY);
+				driveMotor.setPID(0.1, 0.001, 1023.0 / 20660.0);
+			}
 		}
 
 		// configure angle motors
 		for (int i = 0; i < moduleAngleMotors.length; i++) {
-			WPI_TalonFX steeringMotor = moduleAngleMotors[i];
+			MotorController steeringMotor = moduleAngleMotors[i];
 			steeringMotor.configFactoryDefault();
-			steeringMotor.configSelectedFeedbackSensor(
-					TalonFXFeedbackDevice.IntegratedSensor, talonFXLoopNumber, canTimeoutMS);
+			steeringMotor.setNeutralMode(MotorNeutralMode.COAST);
```

Comment:
Last year the steer motors were in brake mode - see https://github.com/SwerveDriveSpecialties/swerve-lib-2022-unmaintained/blob/55f3f1ad9e6bd81e56779d022a40917aacf8d3b3/src/main/java/com/swervedrivespecialties/swervelib/ctre/Falcon500SteerControllerFactoryBuilder.java#L127
This might also be one of the causes of the robot to drift while driving.